app/src/main/kotlin/org/fossify/notes/fragments/TextFragment.kt

```diff
@@ -200,6 +205,29 @@ class TextFragment : NoteFragment() {
         setTextWatcher()
     }
 
+    private fun setupKeyboardListener() {
+        requireActivity().window.decorView.apply {
+            setOnApplyWindowInsetsListener { view, insets ->
+                val windowInsets = WindowInsetsCompat.toWindowInsetsCompat(insets)
+                if (windowInsets.isVisible(WindowInsetsCompat.Type.ime())) {
+                    if (config!!.placeCursorToEnd) {
+                        scrollToEndOfNote()
+                    }
+                    setOnApplyWindowInsetsListener(null) // clear to avoid scrolling every time
+                }
+                view.onApplyWindowInsets(insets)
+            }
+        }
+    }
+
+    private fun scrollToEndOfNote() {
+        val rect = Rect()
+        noteEditText.getFocusedRect(rect)
+        binding.notesScrollview.post {
+            binding.notesScrollview.scrollTo(0, rect.bottom)
+        }
+    }
+
```

Comment:
```suggestion
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupKeyboardListener()
    }

    private fun setupKeyboardListener() {
        ViewCompat.setOnApplyWindowInsetsListener(binding.root) { view, insets ->
            if (insets.isVisible(WindowInsetsCompat.Type.ime())) {
                noteEditText.post {
                    noteEditText.bringPointIntoView(noteEditText.selectionEnd)
                }
            }
            insets
        }
    }
```

You might have to import some things.